### PR TITLE
all is done

### PR DESCRIPTION
**Pull request by ivlytvynets**

### CODE REVIEW

File: src/main/java/com/dev/cinema/lib/PasswordValidator.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    String fieldMatch();
+    String message();
+    String field();
+    Class<?>[] groups() default {};
+    Class<? extends Payload>[] payload() default {};
+    @Target({ ElementType.TYPE })
+    @Retention(RetentionPolicy.RUNTIME)
+    @interface List {
+        PasswordValidator[] value();
+    }
```
Comment (gigbat) [inline src/main/java/com/dev/cinema/lib/PasswordValidator.java]: do we need this?

**Review by gigbat — APPROVED** [review]

File: src/main/java/com/dev/cinema/controller/OrderController.java [hunk 2]
```diff
@@ -32,14 +32,15 @@ public OrderController(OrderMapper orderMapper, OrderService orderService,
     }
 
     @PostMapping("/complete")
-    public void complete(@RequestParam Long userId) {
-        ShoppingCart shoppingCart = shoppingCartService.getByUser(userService.get(userId));
-        orderService.completeOrder(shoppingCart);
+    public void complete(Authentication authentication) {
+        User user = userService.findByEmail(authentication.getName()).get();
+        orderService.completeOrder(shoppingCartService.getByUser(userService.get(user.getId())));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controller/OrderController.java]: can we simplify this line?

File: src/main/java/com/dev/cinema/controller/OrderController.java [hunk 3]
```diff
@@ -32,14 +32,15 @@ public OrderController(OrderMapper orderMapper, OrderService orderService,
...
     }
 
     @GetMapping
-    public List<OrderResponseDto> get(@RequestParam Long userId) {
-        List<Order> orders = orderService.getOrdersHistory(userService.get(userId));
+    public List<OrderResponseDto> get(Authentication authentication) {
+        User user = userService.findByEmail(authentication.getName()).get();
+        List<Order> orders = orderService.getOrdersHistory(userService.get(user.getId()));
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controller/OrderController.java]: same

File: src/main/java/com/dev/cinema/model/dto/UserRequestDto.java [hunk 4]
```diff
@@ -1,8 +1,20 @@
 package com.dev.cinema.model.dto;
 
+import com.dev.cinema.lib.EmailValidator;
+import com.dev.cinema.lib.PasswordValidator;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+
+@PasswordValidator(field = "password",
+        fieldMatch = "repeatPassword",
+        message = "Passwords are different")
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/UserRequestDto.java]:
```suggestion
@PasswordValidator(message = "Passwords are different")
```

File: src/main/java/com/dev/cinema/lib/PasswordValidator.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+import java.lang.annotation.Target;
+
+@Documented
+@Constraint(validatedBy = PasswordValidatorImpl.class)
+@Target({ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface PasswordValidator {
+    String fieldMatch();
+    String message();
+    String field();
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/lib/PasswordValidator.java]:
```suggestion
    String message();
```

File: src/main/java/com/dev/cinema/lib/impl/PasswordValidatorImpl.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+public class PasswordValidatorImpl
+        implements ConstraintValidator<PasswordValidator, UserRequestDto> {
+    @Override
+    public boolean isValid(UserRequestDto userRequestDto,
+                           ConstraintValidatorContext constraintValidatorContext) {
+        return userRequestDto != null && userRequestDto.getPassword() != null
+                && userRequestDto.getPassword().equals(userRequestDto.getRepeatPassword());
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/lib/impl/PasswordValidatorImpl.java:14]: explanation to the previous comments: you do not use these fields in the logic

File: src/main/java/com/dev/cinema/model/dto/UserRequestDto.java [hunk 7]
```diff
@@ -11,4 +23,8 @@ public String getEmail() {
     public String getPassword() {
         return password;
     }
+
+    public String getRepeatPassword() {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/UserRequestDto.java:25]: where is the setter?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/model/dto/UserRequestDto.java (hunk 7 above)
Comment (ivlytvynets) [reply src/main/java/com/dev/cinema/model/dto/UserRequestDto.java:25]: we don't need it

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**